pwr.md

```diff
@@ -0,0 +1,144 @@
+>Available profiles are `default`, `performance`, `balance_performance`, `balance_power` and `power`.
+
+The **cpu** object attributes which can be adjusted by the application are `uncore_max_freq` and `uncore_min_freq`, these set the desired uncore frequency at which an application wants the uncore to operate within.
+>Available frequencies are between `uncore_hw_max` and `uncore_hw_min`.
+
+```python
+core.min_freq = core.lowest_freq
+core.max_freq = core.highest_freq
+
+cpu.uncore_max_freq = cpu.uncore_hw_max
+cpu.uncore_min_freq = cpu.uncore_hw_min
+```
+
+### Committing 
+Modification of the power settings of a system is done by altering the core or CPU characteristics, as shown above, 
+then finalizing with the `commit()` function call, this will commit your changes and write to relevant MSRs or sysfs entries.
+```python 
+cores = pwr.get_cores()
+
+for core in cores:
+    core.min_freq = core.base_freq
+    core.max_freq = core.highest_freq
+    core.commit()
+```
+
+
+### Pre-set Profiles
+When an application is modifying the desired min and max core frequencies, pre-set configurations can also be applied, these will overwrite current configurations and commit the pre-sets.  
+* `minimum`:            Set all cores minimum to 800Mhz and maximum to 800Mhz.
+* `maximum`:            Set all cores minimum to 3900Mhz and maximum to 3900Mhz.
+* `base`:               Set all cores minimum to 2300Mhz and maximum to 2300Mhz.
+* `default`:            Set all cores minimum to 800Mhz and maximum to 3900Mhz.
+* `no_turbo`:           Set all cores minimum to 800Mhz and maximum to 2300Mhz.
+* `sst_bf_base`:        Set high priority cores minimum and maximum to 2700Mhz, normal priority cores minimum and maximum to 2100Mhz.
+* `sst_bf_high_turbo`:  Set high priority cores minimum to 2700Mhz and maximum to 3900Mhz, normal priority cores minimum to 800Mhz and maximum to 2100Mhz.
+* `sst_bf_low_turbo`:   Set high priority cores minimum to 2700Mhz and maximum to 2700Mhz, normal priority cores minimum to 800Mhz and maximum to 3900Mhz.
+>Specific frequencies will depend on system and configuration.
+
+```python
+For c in cores:
+    c.commit("sst_bf_base") # Configure cores with the SST_BF configuration
+```
+
+## Refreshing cpu stats
+On library initialization the systems capabilities and stats are retrieved.
+CPU stats can become out of date, such as `curr_freq` or `sst_bf_configured`. These can be refreshed in with `refresh_stats()`.
+It is *advised* that CPU and core objects be updated with `refresh_stats()` before requesting object data.
+
+core.refresh_stats() will update:
+* `curr_freq`             
+* `min_freq`                
+* `max_freq`                
+* `epp`
+                
+cpu.refresh_stats() will update:
+* `uncore_freq`
+* `uncore_max_freq` 
+* `uncore_min_freq`
+* `sst_bf_configured`
+
+```python
+for c in cores:
+    c.refresh_stats()
+    if c.sst_bf_configured:
+        c.commit("sst_bf_high_turbo")
```

Comment:
This seems a little circular. Perhaps you meant `if not c.sst_bf_configured:`? Otherwise you're configuring SST-BF only if it already was configured...

Reply:
no that should be "sst_bf_enabled"

Reply:
If you wanted to demonstrate core stats changing after a profile commit, perhaps you might want to put `refresh_stats` _after_ commit, not (just) before?

Reply:
no I wasn't demonstrating "core stats changing after a profile commit", I was demonstrating that an application will need to refresh its stats before requesting data.

Rather than before _and_ after any changes.